cardano-node/src/Cardano/CLI/Ops.hs

```diff
@@ -407,45 +402,56 @@ handleMuxError :: MuxError -> IO ()
 handleMuxError err = print err
 
 localInitiatorNetworkApplication
-  :: forall blk m peer.
+  :: forall blk m.
      ( RunNode blk
      , Condense (HeaderHash blk)
-     , IOLike m
      , MonadIO m
+     , MonadST    m
      , MonadTimer m
      )
   => Proxy blk
   -> TopLevelConfig blk
   -> Versions NodeToClientVersion DictVersion
-              (peer -> OuroborosApplication InitiatorApp LB.ByteString m () Void)
+              (LocalConnectionId -> OuroborosApplication InitiatorApp LB.ByteString m () Void)
 localInitiatorNetworkApplication proxy cfg =
-    simpleSingletonVersions
-      NodeToClientV_1
-      (NodeToClientVersionData { networkMagic = nodeNetworkMagic proxy cfg })
-      (DictVersion nodeToClientCodecCBORTerm) $ \_peerid ->
-
-      nodeToClientProtocols
-        NodeToClientProtocols {
-          localChainSyncProtocol =
-            InitiatorProtocolOnly $
-              MuxPeer
-                nullTracer
-                localChainSyncCodec
-                (chainSyncClientPeer chainSyncClient)
-
-        , localTxSubmissionProtocol =
-            InitiatorProtocolOnly $
-              MuxPeer
-                nullTracer
-                localTxSubmissionCodec
-                (localTxSubmissionClientPeer localTxSubmissionClientNull)
-        }
+    foldMapVersions
```

Comment:
Is this `Ouroboros.Network.Protocol.Handshake.Version.foldMapVersions` ?
For me `foldMapVersions` is not a good name.